builder/build.go

```diff
@@ -152,34 +161,41 @@ type dockerBuild struct {
 	BuildLabelMap    map[string]string
 }
 
-// createBuildTemplate creates temporary build folder to perform a Docker build with language template
-func createBuildTemplate(functionName string, handler string, language string, useFunction bool) (string, error) {
+const defaultHandlerFolder = "function"
+
+// createBuildContext creates temporary build folder to perform a Docker build with language template
+func createBuildContext(functionName string, handler string, language string, useFunction bool, handlerFolder string) (string, error) {
 	tempPath := fmt.Sprintf("./build/%s/", functionName)
 	fmt.Printf("Clearing temporary build folder: %s\n", tempPath)
 
 	clearErr := os.RemoveAll(tempPath)
 	if clearErr != nil {
-		fmt.Printf("Error clearing temporary build folder %s\n", tempPath)
+		fmt.Printf("Error clearing temporary build folder: %s\n", tempPath)
 		return tempPath, clearErr
 	}
 
 	functionPath := tempPath
+
 	if useFunction {
-		functionPath += "/function"
+		if handlerFolder == "" {
+			functionPath = path.Join(functionPath, defaultHandlerFolder)
```

Comment:
Not sure how this even worked before given that:
```
functionPath := tempPath    (./build/somename/)
                                             ??
functionPath += "/function" (./build/somename//function)
```
(talking about the old revision)

Reply:
Yes, I've found this kind of code in a few places and I'm fixing it up to use `path.Join`, it was probably written by younger me.

Reply:
Does it make sense to replace all of the string formatting paths with path.Join?   There are a couple more like this one https://github.com/openfaas/faas-cli/pull/755/files#diff-85b0a27de88442108d89cc278530678eR168 and this one https://github.com/openfaas/faas-cli/pull/755/files#diff-85b0a27de88442108d89cc278530678eR345

Reply:
I left those as they were a bit more involved with the sprint, I think path.Join takes a slice?